extensions/indexes/lucene/src/org/exist/indexing/lucene/NodePathPattern.java

```diff
@@ -0,0 +1,230 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package org.exist.indexing.lucene;
+
+//import org.apache.logging.log4j.LogManager;
+//import org.apache.logging.log4j.Logger;
+import org.exist.dom.QName;
+import org.exist.storage.NodePath;
+import org.exist.storage.NodePath2;
+import org.exist.util.FastStringBuffer;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * @author Stanislav Jordanov <stenlee@gmail.com>
+ * @version 1.0
+ *
+ * Class NodePathPattern is a replacement for class NodePath
+ * in cases it was used not as a path, but as a path pattern.
+ * Most notably this mis-design was employed in LuceneConfig and LuceneIndexConfig.
+ *
+ * This is required in order to implement the feature requested/discussed here:
+ *  @see <a href='https://sourceforge.net/p/exist/mailman/message/36392026/'>
+ *      [Exist-open] Are more elaborate xpath expressions allowed in Lucene's index config <text match='...'/>
+ *      </a>
+ *
+ * After class NodePath2 was introduced and replaced NodePath in all cases related to Lucene index
+ * element walking and matching, now all that is left in order to have the desired feature implemented
+ * is implementing properly NodePathPattern.match methods, w/o modifying the originally used NodePath.
+ */
+public class NodePathPattern {
+
+    private final NodePath  qnPath;
+    private final ArrayList<Predicate> predicates = new ArrayList<>();
+
+
+    interface Predicate {
+        boolean  evaluate(NodePath2 nodePath, int elementIdx);
+    }
+
+    private final static Predicate CONST_TRUE_PREDICATE = new Predicate() {
+        @Override
+        public boolean evaluate(NodePath2 nodePath, int elementIdx) {
+            return true;
+        }
+    };
+
+    static class SimpleAttrEqValuePredicate implements Predicate {
+        private final String attrName;
+        private final String attrVal;
+
+        SimpleAttrEqValuePredicate(String attrName, String attrVal) {
+            this.attrName = attrName;
+            this.attrVal  = attrVal;
+        }
+
+        @Override
+        public boolean evaluate(NodePath2 nodePath, int elementIdx) {
+            String val = nodePath.attribs(elementIdx).get(attrName);
+            return val != null ? val.equals(attrVal) : attrVal == null;
+        }
+    }
+
+
+    public NodePathPattern(Map<String, String> namespaces, String matchPattern) {
+        qnPath = new NodePath();
+        parseXPathExpression(namespaces, matchPattern);
+    }
+
+    public NodePathPattern(final QName qname) {
+        qnPath = new NodePath(qname);
+    }
+
+    private void parseXPathExpression(final Map<String, String> namespaces, final String matchPattern) {
+        //TODO : compute better length
```

Comment:
ToDo?

Reply:
NodePathPattern.java is partial copy-paste-edit from NodePath.java
In particular, parseXPathExpression is copy-paste-edit of NodePath's
   private void init(final Map<String, String> namespaces, final String path) {
        //TODO : compute better length
        final FastStringBuffer token = new FastStringBuffer(path.length());

So you see ... I just inherited the TODO from there.
Should have deleted it.
Delete it if you can, please.